rdagent/components/coder/data_science/pipeline/eval.py

```diff
@@ -55,13 +55,50 @@ def evaluate(
 
         env = get_ds_env(extra_volumes={self.scen.debug_path: T("scenarios.data_science.share:scen.input_path").r()})
 
-        # Clean the scores.csv & submission.csv.
-        implementation.execute(env=env, entry=get_clear_ws_cmd())
-        result = implementation.run(env=env, entry=f"python -m coverage run main.py")
-        implementation.running_info.running_time = result.running_time
-        execute_ret_code = result.exit_code
-        stdout = remove_eda_part(result.stdout)
-        stdout += f"The code executed {'successfully' if execute_ret_code == 0 else 'failed'}."
+        stdout = ""
+        run_full_data = True
+        if not DS_RD_SETTING.sample_data:
+            # Because coder runs on full data, we need to run debug mode in advance to save time
+            implementation.execute(env=env, entry=get_clear_ws_cmd())
+            result = implementation.run(env=env, entry=f"python -m coverage run main.py --debug")
+            if result.exit_code != 0:
+                stdout += "Debug mode failed to run. Please check the stdout:\n"
+                stdout += result.stdout
+                run_full_data = False
+            else:
+                stdout += "Debug mode ran successfully.\n"
+                debug_time, full_estimated_time = None, None
+                if match := re.search(r"debug_time:\s*(\d+(?:.\d+)?)", result.stdout, re.DOTALL):
+                    debug_time = float(match.group(1))
+                if match := re.search(r"estimated_time:\s*(\d+(?:.\d+)?)", result.stdout, re.DOTALL):
+                    full_estimated_time = float(match.group(1))
+                if debug_time is not None and full_estimated_time is not None:
+                    stdout += f"Debug mode ran in {debug_time:.2f} seconds, estimated full run time is {full_estimated_time:.2f} seconds.\n"
+                    if full_estimated_time < env.conf.running_timeout_period * 3:
+                        stdout += "The estimated full run time is less than three times the timeout period, proceeding with full data run.\n"
+                    else:
+                        stdout += f"The estimated full run time is more than three times the timeout period, skipping full data run.\nFollowing the stdout of the debug mode run:\n{result.stdout.strip()}\n"
+                        run_full_data = False
+                else:
+                    stdout += (
+                        "Debug mode did not provide debug_time or estimated_time, proceeding with full data run.\n"
+                    )
+
+        if run_full_data:
```

Comment:
I feel running and coding are tighly coupled now (we copy the running code from runner, and then add a branch to skip the running the the runner).
If we have to couple them,  we should share the code at least.